spid_cie_oidc/provider/views.py

```diff
@@ -1,7 +1,240 @@
+        jwk = self.find_jwk(header, jwks)
+        if not jwk:
+            logger.error(
+                f"Invalid jwk for {payload['iss']}. "
+                f"{header['kid']} not found in {jwks}"
+            )
+            raise HttpResponseForbidden()
+
+        try:
+            verify_jws(req, jwk)
+        except Exception as e:
+            logger.error(
+                "Authz request object signature validation failed "
+                f"for {payload['iss']}: {e} "
+            )
+            raise HttpResponseForbidden()
+
+        return payload
+
+
+class AuthzRequestView(OpBase, View):
+    """View which processes the actual Authz request and
+    returns a Http Redirect
+    """
+    template = "op_user_login.html"
+
+    def get(self, request, *args, **kwargs):
+        """
+            authz request object is received here
+            it's validated and a login prompt is rendered to the user
+        """
+        req = request.GET.get('request', None)
+        if not req:
+            logger.error(
+                f"Missing Authz request object in {dict(request.GET)}"
+            )
+            # TODO: render an error template here
+            return HttpResponse(
+                _("Authorization Request not found.")
+            )
+        authz_req = self.validate_authz_request_object(req)
+
+        try:
+            AuthenticationRequestSpid(**payload)
+        except ValidationError:
+            logger.error(
+                "Authz request object validation failed "
+                f"for {authz_req['iss']}: {e} "
+            )
+            raise HttpResponseForbidden()
+
+        # stores the authz request in a hidden field in the form
+        context = {
+            "form": AuthLoginForm(authz_request_object=req)
+        }
+        return render(request, self.template, context)
+
+    def post(self, request, *args, **kwargs):
+        """
+            When the User prompts his credentials
+        """
+        form = AuthLoginForm(request.POST)
+        if not form.is_valid():
+            return render(request, "login_rp.html", {'form': form})
+
+        self.validate_authz_request_object(req)
+
+        # autenticate the user
+
+        #from django.contrib.auth import authenticate
+        # request.POST lo do a dezhi se è valido faccio authenticate
+        # form.cleanedData
+        # user = authenticate(username=username,
+        #                    password=password)
+        # se va male si ritorna form con errore o password errata
+        # se va tutto bene
+        # creo la session con il jwt dal form
+        # creo un altro web point
+        # faccio redirect su user_content
+
+        # store the User session
+        # redirect the user to a consent page
+
+        redirect_uri = payload['redirect_uri']
+        state = payload['state']
+        response = build_error_response(error, error_description, state)
+        return redirect(redirect_uri, **response)
+
+        # check
```

Comment:
I leave your comments to let you work as you require

when we'll finish we have to remove all these

Reply:
Ok